Feature/adaptive learning rate

leoTiez opened the pull request.

leoTiez committed:
Update: Implemented adaptive learnign rate

Learning rate is  now dynamically passed
When the accuracy does not increase for a number of epochs,
it is decreased by multiplying by a constant factor.
If adpative learning factor is set 1, no adaptive learning
is performed.

Updated docstrings

leoTiez committed:
Update: Changed default parameter for adaptive learning factor

leoTiez committed:
Update: Changed hyperparameter of mnist according to changes in cifar

bt-s commented on RCNN_tf.py:
```diff
@@ -373,6 +376,7 @@ def __init__(
                 name='rcl_layer_1'
             )
 
+
```
Please remove, sir.

leoTiez replied:
Yes, sir!

bt-s commented on RCNN_tf.py:
```diff
@@ -530,10 +545,14 @@ def train(self, train_data_feats, train_data_labels, val_data_feats,
                         [self.summaries, self.optimiser, self.cross_entropy],
                         feed_dict={
-                            self.rate_placeholder: 0.2
+                            self.rate_placeholder: 0.2,
```
Should be an input parameter to ```train()```.

leoTiez replied:
Indeed, I will change that!

bt-s reviewed (approved):
Molto bene!

adrianc-a reviewed (approved):
Approved with bas' requested changes.

leoTiez committed:
Update: Passing dropout rate as a parameter

Instead of setting it statically to 0.2, the dropout
rate can be passed as a parameter to the function train
